src/server/locals.js

```diff
@@ -135,35 +151,102 @@ function getLocalScopeForFrame(matchedFrame, callback) {
 
     var locals = response.result;
     matchedFrame.stackLocation.locals = {};
-    for (var local of locals) {
-      matchedFrame.stackLocation.locals[local.name] = getLocalValue(local);
+    var localsContext = {
+      localsObject: matchedFrame.stackLocation.locals,
+      options: options,
+      depth: options.depth
     }
-
-    callback(null);
+    async.each(locals, getLocalValue.bind(localsContext), callback);
   });
 }
 
-function getLocalValue(local) {
-  var value;
+function getLocalValue(local, callback) {
+  var localsObject = this.localsObject;
+  var options = this.options;
+  var depth = this.depth;
+
+  function cb(error, value) {
+    if (error) {
+      // Add the relevant data to the error object,
+      // taking care to preserve the innermost data context.
+      if (!error.rollbarContext) {
+        error.rollbarContext = local;
+      }
+      return callback(error);
+    }
 
-  switch (local.value.type) {
-    case 'undefined': value = 'undefined'; break;
-    case 'object': value = getObjectValue(local); break;
-    case 'array': value = getObjectValue(local); break;
-    default: value = local.value.value; break;
+    if (_.typeName(localsObject) === 'array') {
+      localsObject.push(value);
+    } else {
+      localsObject[local.name] = value;
+    }
+    callback(null);
   }
 
-  return value;
+  if(!local.value) {
+    return cb(null, '[unavailable]');
+  }
+
+  switch (local.value.type) {
+    case 'undefined': cb(null, 'undefined'); break;
+    case 'object': getObjectValue(local, options, depth, cb); break;
+    case 'function': cb(null, getObjectType(local)); break;
+    case 'symbol': cb(null, getSymbolValue(local)); break;
+    default: cb(null, local.value.value); break;
+  }
 }
 
-function getObjectValue(local) {
+function getObjectType(local) {
   if (local.value.className) {
-    return '<' + local.value.className + ' object>'
+    return '<' + local.value.className + ' object>';
   } else {
-    return '<object>'
+    return '<object>';
   }
 }
 
+function getSymbolValue(local) {
+  return local.value.description;
+}
+
+function getObjectValue(local, options, depth, callback) {
+  if(!local.value.objectId) {
+    if ('value' in local.value) {
+      // Treat as immediate value. (Known example is `null`.)
+      return callback(null, local.value.value);
+    }
+  }
+
+  if (!depth) {
```

Comment:
Nit: `depth` is named like an integer, but is treated like a boolean. Can we clarify this with `0 === depth`? (Yoda or not, identical or equality; just intent is to make it look like an integer in the conditional.)